minkindr/include/kindr/minimal/implementation/quat-sim-transform-inl.h

```diff
@@ -0,0 +1,42 @@
+#ifndef KINDR_MINIMAL_IMPLEMENTATION_QUAT_SIM_TRANSFORM_INL_H_
+#define KINDR_MINIMAL_IMPLEMENTATION_QUAT_SIM_TRANSFORM_INL_H_
+
+namespace kindr {
+namespace minimal {
+
+template <typename Scalar>
+QuatSimTransformTemplate<Scalar>::QuatSimTransformTemplate() : scale_A_B_(1.) {}
+
+template <typename Scalar>
+QuatSimTransformTemplate<Scalar>::QuatSimTransformTemplate(
+    const Transform& T_A_B, const Scalar scale_A_B_)
+    : T_A_B_(T_A_B), scale_A_B_(scale_A_B_) {}
+
+template <typename Scalar>
+typename QuatSimTransformTemplate<Scalar>::Vectors
+QuatSimTransformTemplate<Scalar>::operator*(const Vectors& rhs) const {
+  return T_A_B_.transformVectorized(scale_A_B_ * rhs);
+}
+
+template <typename Scalar>
+QuatSimTransformTemplate<Scalar>
+QuatSimTransformTemplate<Scalar>::inverse() const {
+  return QuatSimTransformTemplate<Scalar>(
+      QuatTransformationTemplate<Scalar>(
+          T_A_B_.getRotation().inverse(), -T_A_B_.getRotation().inverseRotate(
+              T_A_B_.getPosition() / scale_A_B_)), 1. / scale_A_B_);
+}
+
+template<typename Scalar>
+std::ostream & operator<<(std::ostream & out,
+                          const QuatSimTransformTemplate<Scalar>& sim_3) {
+  out << "Transform:" << std::endl << sim_3.T_A_B_.getTransformationMatrix() <<
```

Comment:
technically, it should be [R * s, t; 0, 1], but IMO that would obscure the scale (right?). [R,t:0,s] would be wrong - [R,t:0, 1/s] would be correct if translation were applied before scaling:
```
x
y       R t       T(xyz)       T(xyz)/s
z   *   0 s   =   s        =   1
1
```
Or did I misunderstand you?